internal/runners/show/show.go

```diff
@@ -158,24 +162,71 @@ func (s *Show) Run(params Params) error {
 		return locale.WrapError(err, "err_show_secrets", "Could not get secret information")
 	}
 
-	data := outputData{
-		ProjectURL:   projectURL,
-		Namespace:    fmt.Sprintf("%s/%s", owner, projectName),
+	rd := RuntimeDetails{
+		NameSpace:    fmt.Sprintf("%s/%s", owner, projectName),
 		Name:         projectName,
 		Organization: owner,
 		Visibility:   visibilityData(owner, projectName, remoteProject),
-		Commit:       commit,
-		Languages:    languages,
-		Platforms:    platforms,
-		Secrets:      secrets,
-		Events:       events,
-		Scripts:      scripts,
+		LastCommit:   commit,
 	}
 
-	s.out.Print(data)
+	outputData := outputData{
+		ProjectURL:     projectURL,
+		RuntimeDetails: rd,
+		Languages:      languages,
+		Platforms:      platforms,
+		Secrets:        secrets,
+		Events:         events,
+		Scripts:        scripts,
+	}
+
+	s.out.Print(output.NewFormatter(outputData).WithFormat(output.PlainFormatName, output.Suppress))
+
+	plainPrint := func(v interface{}) {
+		s.out.Print(output.NewFormatter(output.Suppress).WithFormat(output.PlainFormatName, v))
+	}
+
+	plainPrint(locale.Tl("show_details_intro", "Here are the details of your runtime environment.\n"))
+	plainPrint(
+		struct {
+			*RuntimeDetails `opts:"verticalTable"`
+		}{&rd},
+	)
+	plainPrint(output.Heading(locale.Tl("state_show_events_header", "Events")))
+	plainPrint(events)
+	plainPrint(output.Heading(locale.Tl("state_show_scripts_header", "Scripts")))
+	plainPrint(formatScripts(scripts))
+	plainPrint(output.Heading(locale.Tl("state_show_platforms_header", "Platforms")))
+	plainPrint(platforms)
+	plainPrint(output.Heading(locale.Tl("state_show_languages_header", "Languages")))
+	plainPrint(languages)
```

Comment:
This is overly complicated. Just tack a `MarshalOutput` onto `outputData` and `out.Print(outputData)`. Then in your marshal function do your plain prints if this is a plain output.

`output.NewFormatter` is intended more for when you want to make a really simple tweak to the output.

Reply:
If we are okay with the `MarshalOutput` function calling output.Print instead of just returning what it should print, that is a fine solution...